common/src/main/java/com/microsoft/identity/common/internal/ui/browser/BrowserSelector.java

```diff
@@ -222,16 +234,37 @@ protected static List<Browser> getBrowsers(@NonNull final Context context,
         return browserList;
     }
 
-    private static boolean isCustomTabsServiceSupported(@NonNull final Context context, @NonNull final PackageInfo packageInfo) {
+    /**
+     * Generates a set of SHA-512, Base64 url-safe encoded signature hashes from the provided
+     * array of signatures.
+     */
+    @NonNull
+    public static Set<String> generateSignatureHashes(@NonNull Signature[] signatures) {
+        Set<String> signatureHashes = new HashSet<>();
+        for (Signature signature : signatures) {
+            try {
+                MessageDigest digest = MessageDigest.getInstance(DIGEST_SHA_512);
+                byte[] hashBytes = digest.digest(signature.toByteArray());
+                signatureHashes.add(Base64.encodeToString(hashBytes, Base64.URL_SAFE | Base64.NO_WRAP));
+            } catch (final NoSuchAlgorithmException e) {
+                throw new IllegalStateException(
```

Comment:
I believe this is ultimately going to be caught at the Command Dispatcher and converted to a base exception?

Reply:
yes